File: app/src/main/java/com/manimaran/wikiaudio/models/WikiToken.java
package com.manimaran.wikiaudio.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class WikiToken {
    @SerializedName("query")
    @Expose
    private Query query;

    public Query getQuery() {
        return query;
    }

    public void setQuery(Query query) {
        this.query = query;
    }


    public static class Query {

        @SerializedName("tokens")
        @Expose
        private TokenValue tokenValue;

        public TokenValue getTokenValue() {
            return tokenValue;
        }

        public void setTokenValue(TokenValue tokenValue) {
            this.tokenValue = tokenValue;
        }
    }

    public static class TokenValue {

        @SerializedName("logintoken")
        @Expose
        private String token;

        public String getToken() {
            return token;
        }

        public void setToken(String token) {
            this.token = token;
        }
    }
}

File: app/src/main/java/com/manimarank/spell4wiki/data/model/WikiToken.kt
package com.manimarank.spell4wiki.data.model

import com.google.gson.annotations.SerializedName

/**
 * Model Class for Wiki Token
 * @property query Query?
 */
data class WikiToken(
    @SerializedName("query")
    var query: QueryToken? = null
)

/**
 * Model Class for Query Object
 * @property tokenValue TokenValue?
 * @constructor
 */
data class QueryToken(
    @SerializedName("tokens")
    var tokenValue: TokenValue? = null
)

/**
 *
 * Model Class for Token Values
 * @property loginToken String?
 * @property csrfToken String?
 * @constructor
 */
data class TokenValue(
    @SerializedName("logintoken")
    var loginToken: String? = null,
    @SerializedName("csrftoken")
    var csrfToken: String? = null
)